### PR TITLE
Increase test coverage to 96.6%

### PR DESCRIPTION
**Pull request by elliotwutingfeng**
## Changes

* Added tests for edge cases.
* Added latest Go versions and upgraded GitHub Actions in CI.

### CODE REVIEW

**Comment by gedw99** [comment]
No e . Would appreciate this going in

File: hashcompute_test.go [hunk 1]
```diff
@@ -124,6 +124,34 @@ func TestNilHashCompute(t *testing.T) {
 	}
 }
 
+func TestExtendHashCompute(t *testing.T) {
```
Comment (corona10) [inline hashcompute_test.go:127]: https://github.com/corona10/goimagehash/blob/c0c66db3601904ae329b4b7d768183a90daca3b1/hashcompute_test.go#L207C21-L329

Is this test not covered by the above tests? If not, can we integrate it?

**Review by corona10 — COMMENTED** [review]
Most of the changes look good. Thank you for your contribution.
I just left a small comment.

Sorry for the late response.

File: hashcompute_test.go (hunk 1 above)
Comment (elliotwutingfeng) [reply hashcompute_test.go:127]: Hi,

`TestExtendHashCompute()` covers 2 corner cases in `ExtPerceptionHash()` from `hashcompute.go` that are not covered by the other tests.

```go
	if imgSize <= 0 || imgSize&(imgSize-1) != 0 {
		return nil, errors.New("width * height should be power of 2") // <---
	}
```

and

```go
	if imgSize%lenOfUnit == 0 {
		phash = make([]uint64, imgSize/lenOfUnit)
	} else {
		phash = make([]uint64, imgSize/lenOfUnit+1) // <---
	}
```

**Review by corona10 — APPROVED** [review]

**Comment by corona10** [comment]
LGTM! Thank you for all your efforts!
